### DISCUSSION TITLE
How to achive automation in dataflow for source ip or destination ip suing IPAM

### DISCUSSION THREAD

**Discussion by nayandj95** [Alef-Burzmali/netbox-data-flows#22] [Q&A] [answered]
Dear Team,

I want to do automation in data flow for source ip and data flow IP Using IPAM,
It is difficult to add every IP manually in source and destination .
I want to use IPAM IP instead of doing manually.
Is there any automation part is there to achieve same.
Like i can do with rest api or with web hook for CURD operations to do automation.

If you have any suggestion or solution for the same please let me know it will help me a lot for my project.

Thank you in advance
Nayan Makwana

**Comment by Alef-Burzmali** [comment]
Hello,
What would be the most helpful to you?
- some easier way to add several IP addresses at the same time in a single Object Alias?
- some easier way to create object aliases based on several IP and assign them to a single data flow?
- being able to automatically add IPs to an Object Alias when they are tagged with a certain tag or based on a custom attribute on the IP?
- or are you thinking of something else?

Being a plugin, I'm limited in how I can interact with native objects (which is the reason behind the object aliases to have many-to-many relationships with several types of objects), so any suggestion on what would help most is welcomed.

Or another way that is already possible: since you are talking about REST API, [NetBox already has one](https://docs.netbox.dev/en/stable/integrations/rest-api/), and the plugin exposes its objects through /api/plugins/data-flows. You will need to create one ObjectAliasTarget for each IP and group them in ObjectAlias

**Comment by kamlesh0606** [comment]
@Alef-Burzmali 
Thanks For Given Reply.

Can Possible, Like this way .
![image](https://github.com/Alef-Burzmali/netbox-data-flows/assets/3240948/36f63caf-d5c3-46b0-8f0e-96d57ebf85ea)
 
Put 3 option , user select any one option (IP Addresses , IP Ranges , ...) on source And Destination then possible to remove "Object Alias" parts.

Otherwise Webhook option is good for manage both side.

 2nd point, Rest Api - Can you provide Add/Edit/Delete Api Detail in Postman Or Document Way. Based on it , i will add Webhook part in Netbox...

FYI @nayandj95

**Comment by Alef-Burzmali** [comment, marked as answer]
If you have a recent version of NetBox (>3.5.0) and the plugin (>0.7.0), you should have the selector widget on the right of the "Add aliased objects" form. This selector can filter by many types of IP attributes, including assigned device or VM, please give it a try and let me know if it fills your need.
![image](https://github.com/Alef-Burzmali/netbox-data-flows/assets/1355939/260db973-7d70-4abb-8956-d2dc4ab2243f)

For REST, NetBox auto-generates a Swagger at `/api/schema/swagger-ui/`, but I've just noticed that I've broken something in the plugin's API.

**Comment by kamlesh0606** [comment]
@Alef-Burzmali  thanks for your support

Is it possible to remove Object Alice parts. And in  data-flow, source and destination have direct selection of ip address ?

FYI @nayandj95

**Comment by Alef-Burzmali** [comment]
No, it is not possible. I want to be able to add IP Prefixes and Ranges, and as I cannot modify the native objects to link them to data flows, I must have an intermediate object to support many-to-many generic relationships.

**Comment by kamlesh0606** [comment]
@Alef-Burzmali  ok . Again thanks for quick response.

Last option to webhook parts.

Api have your side problem ? Or netbox team have problem?

**Comment by Alef-Burzmali** [comment]
I've just released a new version to fix the API bug (it was on my side), it should be good now.

FYI, I'll move that issue to a discussion
